src/log.js

```diff
@@ -191,16 +200,17 @@ export class Log {
       return LogLevel.INFO;
     }
 
-    return this.defaultLevelWithFunc_();
+    return this.defaultLevelWithFunc_(win);
   }
 
   /**
+   * @param {!Window} win
    * @return {!LogLevel}
    * @private
    */
-  defaultLevelWithFunc_() {
+  defaultLevelWithFunc_(win) {
     // Delegate to the specific resolver.
-    return this.levelFunc_(getMode().log, getMode().development);
```

Comment:
It would work, but then testing is much harder to set the value for. Much of this could be dropped in a separate PR if we changed how we tested this; the entire `defaultLevelWithFunc_` only exists as a testing handle.